### PR TITLE
chore: AndroidX migration, Gradle update, dependencies updated

### PR DESCRIPTION
**Pull request by kikoso**
The following PR includes a general update to the latest Android things for the `ConstraintLayoutPerformance` demo:

- Migration to AndroidX.
- Gradle update.
- General update for libraries.
- Some aesthetic code improvements.

### CODE REVIEW

File: ConstraintLayoutPerformance/app/src/androidTest/java/com/example/android/perf/test/PerformanceTest.kt [hunk 1]
```diff
@@ -51,10 +52,10 @@ class PerformanceTest {
         assertThat(launcherPackage, notNullValue())
         device.wait(Until.hasObject(By.pkg(launcherPackage).depth(0)), LAUNCH_TIMEOUT)
 
-        val context = InstrumentationRegistry.getTargetContext()
+        val context = getApplicationContext<Context>()
```
Comment (tikurahul) [inline ConstraintLayoutPerformance/app/src/androidTest/java/com/example/android/perf/test/PerformanceTest.kt:55]: I think you want `getTargetContext` and not the `applicationContext`.

File: ConstraintLayoutPerformance/gradle.properties [hunk 2]
```diff
@@ -15,4 +15,6 @@
 # When configured, Gradle will run in incubating parallel mode.
 # This option should only be used with decoupled projects. More details, visit
 # http://www.gradle.org/docs/current/userguide/multi_project_builds.html#sec:decoupled_projects
-# org.gradle.parallel=true
\ No newline at end of file
+# org.gradle.parallel=true
+android.enableJetifier=true
```
Comment (tikurahul) [inline ConstraintLayoutPerformance/gradle.properties:19]: If we switched to all `androidx` modules do we still need jetifier ?

Comment (kikoso) [reply ConstraintLayoutPerformance/gradle.properties:19]: Shouldn't `android.enableJetifier=true ` be added to convert support libraries of potential dependencies to AndroidX?

Comment (tikurahul) [reply ConstraintLayoutPerformance/gradle.properties]: Not if we don't plan on adding old support libraries.

**Pull request merged**